Core/Relationships/RelationshipResolver.cs

```diff
@@ -127,23 +191,24 @@ public class RelationshipResolver
                 }
 
                 user_requested_mods.Add(module);
-                Add(module, new Relationship.UserRequested());
+                Add(module, new SelectionReason.UserRequested());
             }
 
-            // Now that we've already pre-populated modlist, we can resolve
+            // Now that we've already pre-populated the modlist, we can resolve
             // the rest of our dependencies.
 
-            foreach (CkanModule module in user_requested_mods)
+            foreach (var module in user_requested_mods)
             {
                 log.InfoFormat("Resolving relationships for {0}", module.identifier);
                 Resolve(module, options);
             }
 
             var final_modules = new List<Module>(modlist.Values);
-            final_modules.AddRange(registry.InstalledModules.Select(x => x.Module));
+            
 
             if (!options.without_enforce_consistency)
             {
+                final_modules.AddRange(installed_modules);
```

Comment:
The code could be clearer final_modules is only used in the SanityChecker.EnforceConsistency. Actually adding the modules to the installer is done by this point via calls to Resolve and to Add. 

Thinking about it may be clearer to have Resolve() return the modules to add instead of adding them itself.